### PR TITLE
Component Server fix

### PR DESCRIPTION
**Pull request by aliabid94**
See: https://huggingface.slack.com/archives/C02QZLG8GMN/p1703402300689679?thread_ts=1702901295.299349&cid=C02QZLG8GMN

### CODE REVIEW

File: gradio/components/file_explorer.py [hunk 1]
```diff
@@ -139,10 +139,8 @@ def postprocess(self, value: str | list[str] | None) -> FileExplorerData | None:
         return FileExplorerData(root=root)
 
     @server
-    def ls(self, value=None) -> list[dict[str, str]] | None:
+    def ls(self, _) -> list[dict[str, str]] | None:
```
Comment (abidlabs) [inline gradio/components/file_explorer.py:142]: This change -- making the second argument requried -- is causing the tests to fail

**Review by abidlabs — APPROVED** [review]
LGTM thanks @aliabid94!

File: gradio/routes.py [hunk 2]
```diff
@@ -699,7 +699,12 @@ def component_server(body: ComponentServerBody):
                 block = state[component_id]
             else:
                 block = app.get_blocks().blocks[component_id]
-            fn = getattr(block, body.fn_name)
+            fn = getattr(block, body.fn_name, None)
+            if fn is None or not getattr(fn, "_is_server_fn", False):
```
Comment (akx) [inline gradio/routes.py:703]: `None` wouldn't have the attribute either, so this simplifies to
```suggestion
            if not getattr(fn, "_is_server_fn", False):
```

Comment (aliabid94) [reply gradio/routes.py:703]: makes sense, thanks!

Comment (aliabid94) [reply gradio/routes.py:703]: actually needed for type check :/

**Pull request merged**